core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/FabricUtils.scala

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import com.microsoft.azure.synapse.ml.core.env.StreamUtilities
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import spray.json._
+
+case class TokenServiceConfig(tokenServiceEndpoint: String,
+                              clusterType: String,
+                              clusterName: String,
+                              sessionToken: String)
+
+object TokenServiceConfigProtocol extends DefaultJsonProtocol {
+  implicit val TokenServiceConfigFormat: RootJsonFormat[TokenServiceConfig] = jsonFormat4(TokenServiceConfig)
+}
+
+import TokenServiceConfigProtocol._
+
+object FabricUtils {
+  def getFabricContext: Map[String, String] = {
```

Comment:
i thought you wer going to cache this by maki it a lazy val?

Reply:
Sorry, didn't quite get. You want what to be a lazy val?